File: ejecisios/calculadora.py
# ! es igual not 
# || es igual or
# && es igual and

opcion = "+" # la opcion que teclea el usuario
numero = 10 # valor 1 que ingreso el usuario
numero_dos = 5 # valor 2 que ingreso el usuario

resultado = 0  # resultado que le muestras al usuario de la operacion


"""
Opcion 1 utilizando funciones y condicionales 
 """
def esUnNumero(numero):
    return isinstance(numero, int) # true = si es un numero

def validarNumeros(a, b):
    if (not esUnNumero(a) or not esUnNumero(b)):
        print("Coloca un numero porfavor")
        return True
    return False

def sumar(a, b):
    if validarNumeros(a, b): 
        return 0
    return a + b

def restar(a, b):
    if validarNumeros(a, b):
        return 0
    return a - b
def multiplicar(a, b):
    if validarNumeros(a, b):
        return 0
    return a * b
def dividir(a, b):
    if validarNumeros(a, b):
        return 0
    return a / b
def potencia(a, b):
    if validarNumeros(a, b):
        return 0
    return a ** b

if(opcion == "+"):
    resultado = sumar(numero, numero_dos)
elif(opcion == "-"): # else if en JS
    resultado = restar(numero, numero_dos)
elif(opcion == "*"):
    resultado = multiplicar(numero, numero_dos)
elif(opcion == "/"):
    resultado = dividir(numero, numero_dos)
elif(opcion == "**"):
    resultado = potencia(numero, numero_dos)
else:
    print("opcion invalida")
    exit

print(resultado)



""" Opcion 2
Utilizando clases y diccionario (en JS es un objecto)
 """
class Calculador:
    def __init__(self, uno, dos): # constructor
        self.numero = uno
        self.numerito = dos

    def esUnNumero(numero):
        return isinstance(numero, int) # true = si es un numero

    def validarNumeros(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            print("Coloca un numero porfavor")
            return True
        return False

    def sumar(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            return 0
        return self.numero + self.numerito

    def restar(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            return 0
        return self.numero - self.numerito
    def multiplicar(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            return 0
        return self.numero * self.numerito
    def dividir(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            return 0
        return self.numero / self.numerito
    def potencia(self):
        if (not esUnNumero(self.numero) or not esUnNumero(self.numerito)):
            return 0
        return self.numero ** self.numerito

opcion = "*"
calculadora = Calculador(numero, numero_dos)
operaciones = {
    "+": calculadora.sumar,
    "-": calculadora.restar,
    "*": calculadora.multiplicar,
    "/": calculadora.dividir,
    "**": calculadora.potencia,
}

# print(operaciones["+"]())
resultado = operaciones[opcion]()
print(resultado)

File: ejecisios/calculadora.js
/* 
	Realizar una calculadora que realice operaciones aritmeticas
	+ - * / **
*/

/* Valores del usuario inicial */
// const opcion = "**" // la opcion que teclea el usuario
// const numero = 2 // valor 1 que ingreso el usuario
// const numero_dos = 2 // valor 2 que ingreso el usuario
// 
// let resultado = undefined // resultado que le muestras al usuario de la operacion

/* 
opcion 0

if(opcion === "+"){
	resultado = numero + numero_dos
}else if(opcion === "+"){
	resultado = numero + numero_dos
}else if(opcion === "-"){
	resultado = numero - numero_dos
}
*/

/* 
opcion 1

switch (opcion) {
	case "+":
		resultado = numero + numero_dos
		break;
	case "-":
		resultado = numero - numero_dos
		break;
	default:
		console.error("la opcion no existe")
		return
} 
*/

/* function sumar(a, b) {
	return a + b
}

const sumar2 = (a, b) => a + b */

/* opcion 2 */
function esUnNumero(numero) {
	return !isNaN(numero) && typeof numero === 'number'; // true = si es un numero
}

function validarNumeros(a, b) {
	if (!esUnNumero(a) || !esUnNumero(b)) {
		console.log("Coloca un numero porfavor")
		return true
	}
	return false
}

function sumar(a, b) {
	if (validarNumeros(a, b)) return 0
	return a + b
}

function restar(a, b) {
	if (validarNumeros(a, b)) return 0
	return a - b
}
function multiplicar(a, b) {
	if (validarNumeros(a, b)) return 0
	return a * b
}
function dividir(a, b) {
	if (validarNumeros(a, b)) return 0
	return a / b
}
function potencia(a, b) { // 5, 10
	if (validarNumeros(a, b)) return 0
	return a ** b
}

const operacion = {
	"+": sumar,
	"-": restar,
	"*": multiplicar,
	"/": dividir,
	"**": potencia,
}

/* Utilizando node */
// const existeOpcion = Object.keys(operacion).includes(opcion)
// if (!existeOpcion) {
// 	throw new Error("Opcion invalida")
// 	// console.error("Opcion invalida")
// 	// return 0
// }

// resultado = operacion[opcion](numero, numero_dos)
// // resultado = potencia(5, 10)
// console.log(resultado)

/* Utilizando html */
function devolverNuevosValores() {
	const numero_uno = parseInt(document.getElementById("numero_uno").value)
	const numero_dos = parseInt(document.getElementById("numero_dos").value)
	return [numero_uno, numero_dos]
}

function imprimir_resultado(resultado) {
	document.getElementById("resultado").innerText = resultado
}

const btnSuma = document.querySelector("#suma")
const btnResta = document.querySelector("#resta")
const btnMultiplicacion = document.querySelector("#multiplicacion")




btnSuma.addEventListener("click", () => {
	const [numero_uno, numero_dos] = devolverNuevosValores()
	const resultado = operacion["+"](numero_uno, numero_dos)
	imprimir_resultado(resultado)
})
btnResta.addEventListener("click", () => {
	const [numero_uno, numero_dos] = devolverNuevosValores()
	const resultado = operacion["-"](numero_uno, numero_dos)
	imprimir_resultado(resultado)
})
btnMultiplicacion.addEventListener("click", () => {
	const [numero_uno, numero_dos] = devolverNuevosValores()
	const resultado = operacion["*"](numero_uno, numero_dos)
	imprimir_resultado(resultado)
})